pkg/cmd/channel/create/create.go

```diff
@@ -0,0 +1,187 @@
+		`, constants.ExecutableName),
+		RunE: func(c *cobra.Command, args []string) error {
+			opts := NewCreateOptions(createFlags, cmd.NewDependencies(f, c))
+
+			return createRun(opts)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVarP(&createFlags.Name.Value, createFlags.Name.Name, "n", "", "Name of the channel")
+	flags.StringVarP(&createFlags.Project.Value, createFlags.Project.Name, "p", "", "Project to create channel in")
+	flags.StringVarP(&createFlags.Description.Value, createFlags.Description.Name, "d", "", "Description of the channel")
+	flags.StringVarP(&createFlags.Lifecycle.Value, createFlags.Lifecycle.Name, "l", "", "The lifecycle to use for the channel")
+	flags.BoolVar(&createFlags.Default.Value, createFlags.Default.Name, false, "Set this channel as default")
+
+	return cmd
+}
+
+func createRun(opts *CreateOptions) error {
+	if !opts.NoPrompt {
+		err := PromptMissing(opts)
+		if err != nil {
+			return err
+		}
+	}
+
+	project, err := opts.Client.Projects.GetByIdentifier(opts.Project.Value)
+	if err != nil {
+		return err
+	}
+
+	lifecycle, err := opts.Client.Lifecycles.GetByIDOrName(opts.Lifecycle.Value)
+	if err != nil {
+		return err
+	}
+
+	channel := channels.NewChannel(opts.Name.Value, project.GetID())
+	channel.Description = opts.Description.Value
+	channel.IsDefault = opts.Default.Value
+	channel.LifecycleID = lifecycle.GetID()
+
+	createChannel, err := channels.Add(opts.Client, channel)
+	if err != nil {
+		return err
+	}
+
+	_, err = fmt.Fprintf(opts.Out, "\nSuccessfully created channel '%s', (%s).\n", createChannel.Name, createChannel.GetID())
+	if err != nil {
+		return err
+	}
+
+	link := output.Bluef("%s/app#/%s/projects/%s/deployments/channels/edit/%s", opts.Host, opts.Space.GetID(), project.Slug, createChannel.GetID())
+	fmt.Fprintf(opts.Out, "View this channel on Octopus Deploy: %s\n", link)
+
+	if !opts.NoPrompt {
+		autoCmd := flag.GenerateAutomationCmd(opts.CmdPath, opts.Name, opts.Project, opts.Description, opts.Default, opts.Lifecycle)
+		fmt.Fprintf(opts.Out, "%s\n", autoCmd)
+	}
+
+	return nil
+}
+
+func PromptMissing(opts *CreateOptions) error {
+	err := question.AskName(opts.Ask, "", "channel", &opts.Name.Value)
+	if err != nil {
+		return err
+	}
+
+	var selectedProject *projects.Project
+	if opts.Project.Value == "" {
+		selectedProject, err = selectors.Project("Select the project in which the channel will be created", opts.Client, opts.Ask)
+		if err != nil {
+			return err
+		}
+	} else {
+		selectedProject, err = selectors.FindProject(opts.Client, opts.Project.Value)
+		if err != nil {
+			return err
+		}
+	}
+	opts.Project.Value = selectedProject.Name
+
+	err = question.AskDescription(opts.Ask, "", "channel", &opts.Description.Value)
+	if err != nil {
+		return err
+	}
+
+	var selectedLifecycle *lifecycles.Lifecycle
+	if opts.Lifecycle.Value == "" {
+		selectedLifecycle, err = selectors.Lifecycle("Select the lifecycle to use for the channel", opts.Client, opts.Ask)
```

Comment:
I think this need be modified to support the `Inherit from project` option

Reply:
Yes, you are right. I forgot about that option.